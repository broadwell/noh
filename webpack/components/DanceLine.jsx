import React from "react";
import PropTypes from "prop-types";
import CellDance from "./CellDance";
import { fillGrid } from "../utils";

const DanceLine = props => {
  if (props.grid.length === 0) {
    return <CellDance text="" length={props.length} />;
  }
  const fullData = fillGrid(props.grid, props.length);
  const danceCells = fullData.map((cell, idx) => (
    <CellDance
      text={cell.text}
      length={cell.length}
      key={`danceCell${idx}`} // eslint-disable-line react/no-array-index-key
    />
  ));
  return danceCells;
};

DanceLine.propTypes = {
  grid: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  length: PropTypes.number.isRequired
};

export default DanceLine;
