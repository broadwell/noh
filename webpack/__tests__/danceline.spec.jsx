import React from "react";
import { shallow } from "enzyme";
import DanceLine from "../components/DanceLine";
import phrases from "./__fixtures__/phrases.json";

describe("<DanceLine>", () => {
  it("renders as expected by default", () => {
    const component = shallow(
      <DanceLine grid={phrases.phrases[0].dance.grid} length={13} />
    );
    expect(component).toMatchSnapshot();
  });
  it("renders as expected with no dance data", () => {
    const component = shallow(<DanceLine grid={[]} length={13} />);
    expect(component).toMatchSnapshot();
  });
});
