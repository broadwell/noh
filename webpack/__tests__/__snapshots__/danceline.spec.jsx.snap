// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DanceLine> renders as expected by default 1`] = `
Array [
  <CellDance
    key="danceCell0"
    length={1}
    text=""
  />,
  <CellDance
    key="danceCell1"
    length={4}
    text="9. Kneeling facing front."
  />,
  <CellDance
    key="danceCell2"
    length={8}
    text=""
  />,
]
`;

exports[`<DanceLine> renders as expected with no dance data 1`] = `
<CellDance
  length={13}
  text=""
/>
`;
